src/domain/flow-system/domain/src/repos/flow/pending_flow_repository.rs

```diff
@@ -0,0 +1,32 @@
+// Copyright Kamu Data, Inc. and contributors. All rights reserved.
+//
+// Use of this software is governed by the Business Source License
+// included in the LICENSE file.
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0.
+
+use internal_error::InternalError;
+
+use crate::{FlowID, FlowKey};
+
+////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
+
+#[async_trait::async_trait]
+pub trait PendingFlowRepository: Send + Sync {
```

Comment:
Maybe I'm musunderstanding things, but can't this repo be replaced by the existing `flows` table if it will have `is_pending` field, or as I suggested elsewhere: `where completed_at is null`?

Reply:
I've joint the interfaces. Regarding the implementation proposals, this needs some extra thinking.

Reply:
Okay, implementation expressed as a query to existing tables, `pending_flows` table dropped.